crates/forge/src/collecting.rs

```diff
@@ -0,0 +1,588 @@
+            exit_first: false,
+            fork: vec![],
+            fuzzer_runs: Some(1234),
+            fuzzer_seed: Some(1000),
+        };
+        let config = RunnerConfig::new(
+            Default::default(),
+            None,
+            false,
+            true,
+            Some(100),
+            Some(32),
+            &config_from_scarb,
+        );
+        assert_eq!(
+            config,
+            RunnerConfig {
+                workspace_root: Default::default(),
+                test_name_filter: None,
+                exact_match: false,
+                exit_first: true,
+                fork_targets: vec![],
+                fuzzer_runs: 100,
+                fuzzer_seed: 32,
+            }
+        );
+    }
+
+    #[test]
+    fn collecting_test_crates() {
+        let temp = TempDir::new().unwrap();
+        temp.copy_from("tests/data/simple_package", &["**/*.cairo", "**/*.toml"])
+            .unwrap();
+        let package_path = Utf8PathBuf::from_path_buf(temp.to_path_buf()).unwrap();
+
+        let test_crates = collect_test_crates(
+            &package_path,
+            "simple_package",
+            &package_path,
+            &TempDir::new().unwrap(),
+        )
+        .unwrap();
+
+        assert!(test_crates.contains(&TestCrate {
+            crate_root: package_path,
+            crate_name: "simple_package".to_string(),
+            crate_type: TestCrateType::Lib,
+        }));
+        assert!(test_crates
+            .iter()
+            .any(|tc| tc.crate_name == "tests" && tc.crate_type == TestCrateType::Tests));
+    }
+
+    #[test]
+    fn packing_tests() {
+        let temp = TempDir::new().unwrap();
+        temp.copy_from("tests/data/simple_package", &["**/*.cairo", "**/*.toml"])
+            .unwrap();
+        let package_path = Utf8PathBuf::from_path_buf(temp.to_path_buf()).unwrap();
+        let tests_path = package_path.join("tests");
+
+        let temp_dir = TempDir::new().unwrap();
+        let tests = pack_tests_into_one_file(&temp_dir, &tests_path).unwrap();
+        let virtual_lib_path = tests.crate_root.join("lib.cairo");
+        let virtual_lib_u8_content = std::fs::read(&virtual_lib_path).unwrap();
+        let virtual_lib_content = std::str::from_utf8(&virtual_lib_u8_content).unwrap();
+
+        assert!(virtual_lib_path.try_exists().unwrap());
+        assert!(virtual_lib_content.contains("mod contract;"));
+        assert!(virtual_lib_content.contains("mod ext_function_test;"));
+        assert!(virtual_lib_content.contains("mod test_simple;"));
+        assert!(virtual_lib_content.contains("mod without_prefix;"));
+    }
+
+    fn program_for_testing() -> Program {
+        Program {
+            type_declarations: vec![],
+            libfunc_declarations: vec![],
+            statements: vec![],
+            funcs: vec![],
+        }
+    }
+
+    #[test]
+    #[allow(clippy::too_many_lines)]
+    fn filtering_tests() {
+        let mocked_tests = TestsFromCrate {
```

Comment:
What if has no tests?

Reply:
Tests stay an empty vector, I'll add an unit test for that.